kernel_gateway/mixins.py

```diff
@@ -65,7 +65,7 @@ def prepare(self):
         package.
         """
         server_token = self.settings.get('kg_auth_token')
-        if server_token:
+        if (server_token != '') & (self.request.method != 'OPTIONS'):
```

Comment:
That should be `and` instead of `&`. Python uses keywords for logical operators.
https://docs.python.org/2/library/stdtypes.html#boolean-operations-and-or-not
https://docs.python.org/2/library/stdtypes.html#bitwise-operations-on-integer-types

Reply:
Not sure if `!= ''` is the correct check here. Could server_token be null instead of the empty string?

Reply:
`if server_token and self.request.method != 'OPTIONS':`

Reply:
Also looks like a test is failing because `self.request` does not exist.

```
Traceback (most recent call last):
  File "/home/travis/build/jupyter/kernel_gateway/kernel_gateway/tests/test_mixins.py", line 49, in test_no_token_required
    self.mixin.prepare()
  File "/home/travis/build/jupyter/kernel_gateway/kernel_gateway/mixins.py", line 68, in prepare
    if (server_token != '') & (self.request.method != 'OPTIONS'):
AttributeError: 'TestableTokenAuthHandler' object has no attribute 'request'
```